API/models/model.go

```diff
@@ -815,8 +941,15 @@ func GetEntitiesUsingAncestorNames(ent string, id primitive.ObjectID, ancestry [
 	return nil, ""
 }
 
-func GetEntityUsingAncestorNames(ent string, id primitive.ObjectID, ancestry []map[string]string) (map[string]interface{}, string) {
-	top, e := GetEntity(bson.M{"_id": id}, ent, u.RequestFilters{})
+func GetEntityUsingAncestorNames(ent string, id primitive.ObjectID, req map[string]interface{}, ancestry []map[string]string) (map[string]interface{}, string) {
+
+	newReq := req
+	if newReq == nil {
+		newReq = bson.M{"_id": id}
+	} else {
+		newReq["_id"] = id
+	}
```

Comment:
rather than adding the id to req, shouldnt we remove the parameter id, and simply check that it is already part of req ? so that the signature would be more like the GetEntity function one

Reply:
then I don't mind